src/artrace.py

```diff
@@ -1350,159 +1054,144 @@ def model_tilt(slf, det, msarc, guesstilts=None, censpec=None, plotQA=False, ref
+            # Perform a PCA on the tilts
+            msgs.info("Performing a PCA on the tilts")
+            ordsnd = np.linspace(0.0, 1.0, msarc.shape[0])
+            xcen = xv[:, np.newaxis].repeat(msarc.shape[0], axis=1)
+            fitted, outpar = arpca.basis(xcen, tiltval, tcoeff, lnpc, ofit, weights=None,
+                                         x0in=ordsnd, mask=maskrw, skipx0=False,
+                                         function=slf._argflag['trace']['orders']['function'])
+            arpca.pc_plot(slf, outpar, ofit, pcadesc="Spectral Tilts PCA", addOne=False)
+            # Extrapolate the remaining orders requested
+            orders = np.linspace(0.0, 1.0, msarc.shape[0])
+            extrap_tilt, outpar = arpca.extrapolate(outpar, orders, function=slf._argflag['trace']['orders']['function'])
+            polytilts = extrap_tilt.T
+        else:
+            # Fit the model with a 2D polynomial
+            msgs.warn("Could not perform a PCA when tracing the spectral tilt" + msgs.newline() +
+                      "Not enough well-traced arc lines")
+            msgs.info("Fitting tilts with a low order, 2D polynomial")
+            wgd = np.where(xtilt != maskval)
+            coeff = arutils.polyfit2d_general(xtilt[wgd], ytilt[wgd], mtilt[wgd], fitxy)
+            polytilts = arutils.polyval2d_general(coeff, np.linspace(0.0, 1.0, msarc.shape[1]),
+                                                  np.linspace(0.0, 1.0, msarc.shape[0]))
+
+        # Now apply the starting guess for the tilts
         if guesstilts is not None:
-#            wfit = np.where(wtilt!=maskval)
-#            coeff = arutils.polyfit2d(xtilt[wfit], ytilt[wfit], ztilt[wfit], fitxy)#, w=wtilt[wfit]/np.max(wtilt[wfit]))
-#            tilts = arutils.polyval2d(coeff, np.arange(msarc.shape[1])/(msarc.shape[1]-1.0), np.arange(msarc.shape[0])/(msarc.shape[0]-1.0))
-            tilts += guesstilts
-    else:
-        msgs.error("Please use trace+tilts+fit1d")
-
-    # Perform a high order 2D polynomial fit to the best PCA tilts, then interpolate
-#    if refine_tilts:
-#        debugger.set_trace()
-
-    #xx, yy = np.meshgrid(np.arange(msarc.shape[1])/(msarc.shape[1]-1.0), np.arange(msarc.shape[0])/(msarc.shape[0]-1.0))
-    #coeff = arutils.polyfit2d(xx, yy, tilts, [8,8])
-    #tilts = arutils.polyval2d(coeff, np.arange(msarc.shape[1])/(msarc.shape[1]-1.0), np.arange(msarc.shape[0])/(msarc.shape[0]-1.0))
-    # Plot the model tilts
-
-#guesstilts[arcdet[j],ordcen[arcdet[j],0]-k]*float(msarc.shape[0]-1.0)
-
-
-    tiltsplot = tilts[arcdet,:].T
+            polytilts += guesstilts
+
+        if slf._argflag['trace']['orders']['tilts'].lower() in ["spline", "interp"]:
+            xspl = np.linspace(0.0, 1.0, msarc.shape[1])
+            yspl = np.append(0.0, np.append(arcdet[np.where(aduse)]/(msarc.shape[0]-1.0), 1.0))
+            zspl = np.zeros((msarc.shape[1], np.sum(aduse)+2))
+            zspl[:, 1:-1] = mtilt[:, np.where(aduse)[0]]
```

Comment:
On KastR there are some lines which are giving junk due to a small number of bad values in ytfit which is wrecking 
arcyutils.polyfit_scan_intext

Either that Cython algorithm needs to include some rejection or we may need to avoid it.